new generate target

ibrokethecloud opened the pull request:
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
<!-- Explain the problem you are aiming to resolve in this PR. -->
PR introduces a new `generate` target for Make which allows `go generate` and `openapi-gen` to be executed via Dapper in a consistent build environment.

**Solution:**
<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->

**Related Issue:**
https://github.com/harvester/harvester/issues/3375
**Test plan:**
<!-- Make sure tests pass on the Circle CI. -->

ibrokethecloud committed:
add code gen as a target

votdev reviewed (approved):
LGTM (taking my comment into account)

votdev committed:
Various changes

- Add `./api` to `DAPPER_OUTPUT` env variable to get the latest `swagger.json`
- Add `./deploy` to `DAPPER_OUTPUT` env variable to copy the generated CRDs back to the host
- Remove duplicate call of the `generate-openapi` script. This is already called in `main.go` via `go:generate` annotation
- Rebase branch

Signed-off-by: Volker Theile <vtheile@suse.com>

FrankYang0529 reviewed (commented):
Overall LGTM. For failed CI pipeline, I think we can fix it by rebasing `master` branch.

FrankYang0529 reviewed (approved):
LGTM. Thanks @ibrokethecloud and @votdev.
